### PR TITLE
Issue #7892: Improve clean up of .ci-temp directory

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #7892: Improve clean up of .ci-temp directory

### CODE REVIEW

File: .ci/wercker.sh [hunk 1]
```diff
@@ -182,7 +189,7 @@ no-error-contribution)
      -Dcheckstyle.configLocation=../../../config/checkstyle_checks.xml
   cd ../
   cd ../../
-  rm -rf checkstyle
+  removeFolderWithProtectedFiles checkstyle
```
Comment (pbludov) [inline .ci/wercker.sh]: This should be `contribution`

Comment (nmancus1) [reply .ci/wercker.sh]: Fixed, but wercker is failing.  Could you please share the log with me?

Comment (pbludov) [reply .ci/wercker.sh]: Now it fails with `find: ‘blog-sonar-build-breaker’: No such file or directory`

**Comment by pbludov** [comment]
@nmancus1 It looks like the folder name is `source`:

```
Cloning into 'blog-sonar-build-breaker'...
Sonar Qualitygate is OK
$ pwd
/pipeline/source
$ ls -l
total 236
-rw-r--r--. 1 root root   2148 Mar 28 07:52 Jenkinsfile
-rw-r--r--. 1 root root  26461 Mar 28 07:52 LICENSE
-rw-r--r--. 1 root root  11358 Mar 28 07:52 LICENSE.apache20
-rw-r--r--. 1 root root   7491 Mar 28 07:52 README.md
-rw-r--r--. 1 root root   1228 Mar 28 07:52 RIGHTS.antlr
-rw-r--r--. 1 root root   2915 Mar 28 07:52 appveyor.yml
-rw-r--r--. 1 root root   2318 Mar 28 07:52 buddy.yml
-rw-r--r--. 1 root root     96 Mar 28 07:52 codeship-services.yml
-rw-r--r--. 1 root root    391 Mar 28 07:52 codeship-steps.yml
drwxr-xr-x. 2 root root   4096 Mar 28 07:52 config
-rw-r--r--. 1 root root 126989 Mar 28 07:52 pom.xml
-rwxr-xr-x. 1 root root   5230 Mar 28 07:52 release.sh
-rw-r--r--. 1 root root   1890 Mar 28 07:52 shippable.yml
drwxr-xr-x. 7 root root   4096 Mar 28 07:52 src
drwxr-xr-x. 9 root root   4096 Mar 28 08:21 target
-rw-r--r--. 1 root root  10921 Mar 28 07:52 wercker.yml
$ cd ..
$ ls -l
total 128
drwxr-xr-x. 70 root root 4096 Mar 28 07:52 cache
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 output
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 script-04e8ab57-b4f8-48a5-84df-d01d132fd78b
drwxr-xr-x.  2 root root 4096 Mar 28 08:20 script-1093f238-db91-4209-8e7f-fb78b83b32e6
drwxr-xr-x.  2 root root 4096 Mar 28 08:14 script-1375cf86-df23-40b9-b0cd-1bc0bf4a8937
drwxr-xr-x.  2 root root 4096 Mar 28 08:12 script-14ec2add-d959-4cad-8065-97c8945dca52
drwxr-xr-x.  2 root root 4096 Mar 28 08:03 script-14fe3502-1717-4ddd-a006-be2400294710
drwxr-xr-x.  2 root root 4096 Mar 28 07:56 script-2324a0fa-0489-4bbe-bd76-d754438beae4
drwxr-xr-x.  2 root root 4096 Mar 28 08:00 script-23704608-c5e0-4b46-8455-787f6a5ab456
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 script-2386d89a-049d-4d13-a176-a3e7c4ce7da7
drwxr-xr-x.  2 root root 4096 Mar 28 08:19 script-451d7a58-8f54-4afb-9b82-dcfdca261bc4
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 script-4763b521-1666-443e-8e08-7a2f98eda9ba
drwxr-xr-x.  2 root root 4096 Mar 28 07:57 script-485eab7f-b137-474b-800d-4c489ce5a55d
drwxr-xr-x.  2 root root 4096 Mar 28 07:53 script-5c0abd04-663d-4400-83d0-f56ea15d26d7
drwxr-xr-x.  2 root root 4096 Mar 28 08:09 script-5e98e632-2a05-418a-bfec-416f43353639
drwxr-xr-x.  2 root root 4096 Mar 28 08:02 script-6770af27-6013-4017-bbb0-df26c4295e0d
drwxr-xr-x.  2 root root 4096 Mar 28 08:02 script-74df0788-bdf9-4908-bdf1-a80015d3c7f9
drwxr-xr-x.  2 root root 4096 Mar 28 08:05 script-81c7174b-1cdb-4086-ab5c-5cf22542acd2
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 script-898647e4-cb3a-4d4d-8f45-42aa87763c38
drwxr-xr-x.  2 root root 4096 Mar 28 08:10 script-8d01f126-b96f-4cb7-935c-7698b6210fd9
drwxr-xr-x.  2 root root 4096 Mar 28 08:00 script-8d815b6c-f5a0-421b-9a29-47471a26c75b
drwxr-xr-x.  2 root root 4096 Mar 28 08:01 script-9d5af3a2-0a2e-431a-adca-84d612a8bff5
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 script-9f3d6b8c-b347-4769-9c98-41988fd9cfbe
drwxr-xr-x.  2 root root 4096 Mar 28 07:54 script-ba0faa01-f386-49a5-856d-b7cf1e862fc3
drwxr-xr-x.  2 root root 4096 Mar 28 07:55 script-bef67557-c152-461d-b2d3-37347a2cb1ca
drwxr-xr-x.  2 root root 4096 Mar 28 08:17 script-d6ff0df9-92da-44d4-95b3-ce73fbbf4389
drwxr-xr-x.  2 root root 4096 Mar 28 08:20 script-e1ed56a6-bdd0-4513-bfc5-9b7ffd98d4ce
drwxr-xr-x.  2 root root 4096 Mar 28 08:07 script-e5031720-f5b6-49e7-959f-9b3c841c74c1
drwxr-xr-x.  2 root root 4096 Mar 28 08:06 script-ec332f28-a40a-41d7-b60d-7d41b64e535f
drwxr-xr-x.  2 root root 4096 Mar 28 08:01 script-f09735f0-b034-4ee4-8207-476a82c0f06f
drwxr-xr-x. 11 root root 4096 Mar 28 08:20 source
drwxr-xr-x.  2 root root 4096 Mar 28 07:52 wercker-init-f6bab49b-1a03-48a3-b07b-1ecc3fe25f80
$ find -delete blog-sonar-build-breaker
find: ‘blog-sonar-build-breaker’: No such file or directory
```

**Comment by pbludov** [comment]
Now it fails on the last step:
image

**Comment by nmancus1** [comment]
@pbludov what are the wercker failures? I can't run `/.ci/wercker.sh sonarqube` to completion locally to determine the directory structure so that I can make sure the script ios in the right place before running `find -delete $1`.  This is where I got the initial directory name from: https://github.com/checkstyle/checkstyle/pull/7880#issuecomment-600109654

**Comment by pbludov** [comment]
@nmancus1 finally got it. There is `cd ..` in the script `sonar_break_build.sh`

this should work:
`removeFolderWithProtectedFiles .ci-temp/blog-sonar-build-breaker`

**Comment by nmancus1** [comment]
@pbludov I removed the `cd..` in sonarqube too, that seemed to do the trick.  Thanks for your help!

File: .ci/wercker.sh (hunk 1 above)
Comment (nmancus1) [reply .ci/wercker.sh]: Fixed.

**Comment by romani** [comment]
@nmancus1 , please rebase, there is conflict.

**Comment by romani** [comment]
@pbludov , please resolve conversation if you are satisfied.

**Comment by romani** [comment]
It would be awesome to do verification of clean folder at the end, to make sure we will not make problems again.

**Pull request merged**